### PR TITLE
Bugfixes for indirect permissions

### PR DESCRIPTION
**Pull request by StefanMaron**

### CODE REVIEW

**Comment by StefanMaron** [comment]
@Arthurvdv can you have a look why the builds are failing for those old versions here?

**Comment by Arthurvdv** [comment]
The problem is caused by namespaces, which aren't there in the beginning of the v12.0 versions of AL.

You need to wrap these with a  preprocessor directive, for example see rule 0039:

https://github.com/StefanMaron/BusinessCentral.LinterCop/blob/976b45b04ce4403cbe26763952ad57e274300585/BusinessCentral.LinterCop/Design/Rule0039ArgumentDifferentTypeThenExpected.cs#L148-L151

File: BusinessCentral.LinterCop/Design/Rule0068CheckObjectPermission.cs [hunk 1]
```diff
@@ -213,20 +214,20 @@ private void CheckProcedureInvocation(IPropertySymbol objectPermissions, string
                 var objectName = typeAndObjectName[typeEndIndex..].Trim().Trim('"');
 
                 // Match against the parameters of the procedure
-                if (type == "tabledata" && objectName == variableTypeName.ToLowerInvariant())
+                if (type == "tabledata" && (objectName == variableType.Name.ToLowerInvariant() || objectName.Replace("\"","") == (variableType.OriginalDefinition.ContainingNamespace.QualifiedName.ToLowerInvariant() + "." + variableType.Name.ToLowerInvariant())))
```
[the remark this answers is not in the document]
Comment (StefanMaron) [reply BusinessCentral.LinterCop/Design/Rule0068CheckObjectPermission.cs:217]: at this point its just a string, so I dont think that method is available

**Comment by StefanMaron** [comment]
@Arthurvdv  can I use the directives in tests as well?

File: BusinessCentral.LinterCop/Design/Rule0068CheckObjectPermission.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Arthurvdv) [reply BusinessCentral.LinterCop/Design/Rule0068CheckObjectPermission.cs]: It's a stringExtension, so I should work

**Pull request merged**